script/openchange_user_cleanup.py

```diff
@@ -63,20 +64,60 @@ def active_openchange_users(self):
 
 
 class ImapCleaner(object):
-    def __init__(self, imap_host="127.0.0.1", imap_port=143, dry_run=False,
-                 samba_helper=SambaOCHelper()):
-        self.imap_host = imap_host
-        self.imap_port = imap_port
+    def __init__(self, dry_run=False, samba_helper=SambaOCHelper()):
         self.dry_run = dry_run
         self.samba_helper = samba_helper
+        self.system_defaults_file = "/etc/sogo/sogo.conf"
+        self.user_defaults_file = \
+            os.path.expanduser("~sogo/GNUstep/Defaults/.GNUstepDefaults")
+
+    def _get_connection_url(self):
+        connection_url = None
+        # read defaults from defaults files
+        # order is important, user defaults must have precedence
+        for f in [self.system_defaults_file, self.user_defaults_file]:
+            if os.path.exists(f):
+                p1 = subprocess.Popen(["sogo-tool", "dump-defaults", "-f", f],
+                                      stdout=subprocess.PIPE)
+                p2 = subprocess.Popen(["awk", "-F\"",
+                                       "/ SOGoIMAPServer =/ {print $2}"],
+                                      stdin=p1.stdout, stdout=subprocess.PIPE)
+                p1.stdout.close()  # Allow p1 to receive a SIGPIPE if p2 exits.
+                tmp = p2.communicate()[0]
+                if tmp:
+                    connection_url = tmp
+
+        return connection_url
 
     def cleanup(self, username, password=""):
         print "===== IMAP cleanup ====="
-        client = imaplib.IMAP4(self.imap_host, self.imap_port)
+
+        connection_url = self._get_connection_url()
```

Comment:
Can you instead provide a fallback mechanism to self._get_connection_url() is no imap_host/post is provided?